configure.in

```diff
@@ -164,7 +164,9 @@ CITUSAC_PROG_CC_CFLAGS_OPT([-Wmissing-format-attribute])
 CITUSAC_PROG_CC_CFLAGS_OPT([-Wmissing-declarations])
 CITUSAC_PROG_CC_CFLAGS_OPT([-Wmissing-prototypes])
 CITUSAC_PROG_CC_CFLAGS_OPT([-Wshadow])
-CITUSAC_PROG_CC_CFLAGS_OPT([-Werror=vla])  # visual studio does not support these
+CITUSAC_PROG_CC_CFLAGS_OPT([-Wvla])  # visual studio does not support these
+CITUSAC_PROG_CC_CFLAGS_OPT([-Werror])
```

Comment:
This is _also_ a topic with a lot of history. See for instance, this comment: https://github.com/citusdata/citus/pull/697#discussion_r73419699

Basically, we want to incentivize using `Werror` internally and in CI, but will _not_ actually make it the default in `configure`, because we don't know everywhere this code will be compiled. In particular, we don't want to break PGDG builds against e.g. upcoming Debian/Ubuntu releases due to new warnings; sure, we'd want to address them once those OS targets cool down/are released, but at that point we'd see the errors _ourselves_ on our own machines compiled with `Werror`.

So, I'm not sure this should be accepted. The fixes, yes. The `configure` changes, no.